MekHQ/src/mekhq/campaign/rating/CampaignOpsReputation.java

```diff
@@ -503,22 +503,20 @@ public int getTransportValue() {
         TransportCapacityIndicators tci = new TransportCapacityIndicators();
         tci.updateCapacityIndicators(getMechBayCount(), getMechCount());
         tci.updateCapacityIndicators(getProtoBayCount(), getProtoCount());
-        tci.updateCapacityIndicators(getHeavyVeeBayCount(), getHeavyVeeCount());
+        tci.updateCapacityIndicators(getBaBayCount(), getBattleArmorCount() / 5); // battle armor bays can hold 5 battle armor per bay
+        tci.updateCapacityIndicators(getInfantryBayCount(), calcInfantryPlatoons());
 
-        // vehicles are the only units that can share heavy/light bays so we have some special logic
+        // vehicles can share heavy/light bays, while small craft and fighters can share bays
         // we've stuffed all possible heavy vehicles into heavy vehicle bays.
         // if we have some heavy vehicle bays left over, add them to the light vehicle bay count
-        int heavyVeeBays = getHeavyVeeBayCount() - getHeavyVeeCount();
-        int lightVeeBays = getLightVeeBayCount();
-        if(heavyVeeBays > 0) {
-            lightVeeBays += heavyVeeBays;
-        }
+        // The same is done for small craft and fighters, just replace heavy vehicle with small craft and light vehicle with fighters
+        int excessHeavyVeeBays = Math.max(getHeavyVeeBayCount() - getHeavyVeeCount(), 0);
+        int excessSmallCraftBays = Math.max(getSmallCraftBayCount() - getSmallCraftCount(), 0);
 
-        tci.updateCapacityIndicators(lightVeeBays, getLightVeeCount());
-        tci.updateCapacityIndicators(getFighterBayCount(), getFighterCount());
-        tci.updateCapacityIndicators(getBaBayCount(), getBattleArmorCount() / 5); // battle armor bays can hold 5 battle armor per bay
-        tci.updateCapacityIndicators(getInfantryBayCount(), calcInfantryPlatoons());
-        tci.updateCapacityIndicators(getSmallCraftBayCount(), getSmallCraftCount());
+        tci.updateCapacityIndicators(getLightVeeBayCount() + excessHeavyVeeBays, getLightVeeCount());
+        tci.updateCapacityIndicators(getHeavyVeeBayCount() - excessHeavyVeeBays, getHeavyVeeCount());
```

Comment:
Not quite sure why we're subtracting excessHeavyVeeBays here.

Reply:
I was testing around a bug fix, and forgot that I didn't finish the code (and missed it on my own review). Currently we've got a slight logic issue, which is that we count all 'excess' heavy vehicle bays as empty, even if they are filled by light vehicles. So that puts the calculations off.

The fix is an annoying to see one, you subtract the excess bays then add in what bays would be left free. I'll implement that now.

Reply:
Very much an edge case, but might as well fix it while I'm working on that code

Reply:
Although please do let me know if I'm misunderstanding the pre-existing code and this is handled as is.

Reply:
This logic should fix that:
`int heavyVeeBaysFilledByLights;
int excessLightVees = Math.max(getLightVeeCount() - getLightVeeBayCount(), 0);
if (excessLightVees <= excessHeavyVeeBays){
    heavyVeeBaysFilledByLights = excessLightVees;
} else {
    heavyVeeBaysFilledByLights = excessHeavyVeeBays;
}`

Reply:
Not sure I agree.

The basic idea should be as follows:
1. Load all light vehicles into light bays and heavy vehicles into heavy bays. Any light vehicles left over are "excess". That's the second line of the proposed logic.

2. If you have excess heavy vehicle bays and excess light units, load the light units into the heavy bays. I'm not sure that the if block fulfills that condition.  I'm not sure that if block fulfills that condition.

You probably want something like:
updateCapacityIndicators(heavyBays, heavyVeeCount)
updateCapacityIndicators(lightBays + max(heavyBays - heavyVeeCount, 0), lightVeeCount)

and similar for small craft/fighters. That way, you stuff the heavy bays with appropriate units first, then any left overs can be used for light vees.

Reply:
That block would be added before:
```
        tci.updateCapacityIndicators(getLightVeeBayCount() + excessHeavyVeeBays, getLightVeeCount());
        tci.updateCapacityIndicators(getHeavyVeeBayCount() - heavyVeeBaysFilledByLights, getHeavyVeeCount());
        tci.updateCapacityIndicators(getFighterBayCount() + excessSmallCraftBays, getFighterCount());
        tci.updateCapacityIndicators(getSmallCraftBayCount() - smallCraftBaysFilledByFighters, getSmallCraftCount());
```

The difference is that this removes any filled heavy bays from being counted as free by the comparison logic. Same basic logic that follows what you are saying, but with the addition of a catch to stop it from counting bays filled by lights as empty.

Reply:
That makes the whole coding block:
```
       // Light vehicles can use heavy or light vehicle bays, while fighters can use fighter or small craft bays
        // Once we've stuffed stuffed all possible heavy vehicles into heavy vehicle bays.
        // if we have some heavy vehicle bays left over, add them to the light vehicle bay count, and then calculate the
        // number of heavy vehicle bays that are still empty.
        // The same is done for small craft and fighters, just replace heavy vehicle with small craft and light vehicle with fighters
        int excessHeavyVeeBays = Math.max(getHeavyVeeBayCount() - getHeavyVeeCount(), 0);
        int excessSmallCraftBays = Math.max(getSmallCraftBayCount() - getSmallCraftCount(), 0);

        // We need to subtract any filled bays from the count. This follows the following logic:
        // Assume you have 2 heavy vehicle bays, and 4 light vehicle bays, and are trying to store 1 heavy and 5 light vehicles
        // You have 1 more light vehicle than light vehicle bays to store them in, so you check how many free heavy vehicle bays
        // there are. Finding 1, you can store the light vehicle there, and it doesn't count as having excess
        int heavyVeeBaysFilledByLights;
        int excessLightVees = Math.max(getLightVeeCount() - getLightVeeBayCount(), 0);
        if (excessLightVees <= excessHeavyVeeBays){
            heavyVeeBaysFilledByLights = excessLightVees;
        } else {
            heavyVeeBaysFilledByLights = excessHeavyVeeBays;
        }

        int smallCraftBaysFilledByFighters;
        int excessFighters = Math.max(getFighterCount() - getFighterBayCount(), 0);
        if (excessFighters <= excessSmallCraftBays){
            smallCraftBaysFilledByFighters = excessFighters;
        } else {
            smallCraftBaysFilledByFighters = excessSmallCraftBays;
        }

        tci.updateCapacityIndicators(getLightVeeBayCount() + excessHeavyVeeBays, getLightVeeCount());
        tci.updateCapacityIndicators(getHeavyVeeBayCount() - heavyVeeBaysFilledByLights, getHeavyVeeCount());
        tci.updateCapacityIndicators(getFighterBayCount() + excessSmallCraftBays, getFighterCount());
        tci.updateCapacityIndicators(getSmallCraftBayCount() - smallCraftBaysFilledByFighters, getSmallCraftCount());
```

Reply:
Ok, I worked through a couple of examples, and I think I get it. Sounds good. Just put a comment in there, explaining in a little more detail what's going on.